modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/ByteBufStreamInput.java

```diff
@@ -132,7 +132,11 @@ public int skipBytes(int n) throws IOException {
 
     @Override
     public byte readByte() throws IOException {
-        return buffer.readByte();
+        try {
+            return buffer.readByte();
+        } catch (IndexOutOfBoundsException ex) {
+            throw new EOFException();
```

Comment:
can you put the ex as a cause in there?

Reply:
I added it. The ctor does not take it. But I added it use the init method.